pkg/report/template.go

```diff
@@ -46,6 +46,16 @@ func NewTemplateWriter(output io.Writer, outputTemplate string) (*TemplateWriter
 		}
 		return escaped.String()
 	}
+	templateFuncMap["makeRuleMap"] = func() map[string]int {
+		return make(map[string]int)
+	}
+	templateFuncMap["indexRule"] = func(rules map[string]int, vulnerabilityID string) bool {
+		if _, ok := rules[vulnerabilityID]; !ok {
+			rules[vulnerabilityID] = len(rules) + 1
```

Comment:
[sarif validator](https://sarifweb.azurewebsites.net/Validation) shows an error:
```
SARIF1009: runs[0].results[0]: This 'result' object contains a property 'ruleIndex' with value 1,
but 'runs[0].tool.driver.rules' has fewer than 2 elements. 
An index-valued properties must be valid for the array that it refers to.
```
it seems that we should remove `+1`, right?

Reply:
Ah yes, my bad! 😓